distributor/cmd/main.go

```diff
@@ -442,39 +442,60 @@ func pollHTTPEventSource(endpoint string, token string, topics []string) {
 	}
 }
 
+// pollEventsForTopic polls .triggered events from the Keptn api, and forwards them to the receiving service
 func pollEventsForTopic(endpoint string, token string, topic string) {
 	logger.Infof("Retrieving events of type %s", topic)
 	events, err := getEventsFromEndpoint(endpoint, token, topic)
 	if err != nil {
 		logger.Errorf("Could not retrieve events of type %s from endpoint %s: %v", topic, endpoint, err)
 	}
 	logger.Infof("Received %d new .triggered events", len(events))
+
+	// ensure CloudEvents Cache exists
+	if ceCache == nil {
+		logger.Debug("Cache containing sent CloudEvents is nil. Creating a new one")
+		ceCache = lib.NewCloudEventsCache()
+	}
+
+	// iterate over all events, discard the event if it has already been sent
 	for index, _ := range events {
```

Comment:
I know, it's not part of the PR, but could you please replace line 461 and 462 with
```
	for index := range events {
		event := *events[index]
```

Reply:
Will do.